### ISSUE TITLE
Machine Learning Pipeline Issues

### ISSUE THREAD

**Issue by eric15342335** [eric15342335/maimai-dx-regression#1]
- [ ] Remove linear baseline since it is essentially not predicting anything better than the mean.
- [ ] Comment out collinearity removal. Tree-based models (random forest) are safe from that.
- [ ] Implement a proper time series split instead of `train_test_split`. However, since the timestamp is not available, if we use `Chart Constant` to split, then according to the EDA image (`Achv` vs `Chart Constant`), it is essentially impossible to predict the scores because performance drops suddenly, and it is more likely to contain out-of-distribution `Chart Constant` values in the validation set from the model (being trained on the training set)'s point of view.
- [ ] Try stacking (stacked generalization) with RF + LightGBM, then use a non-linear least-squares learner, or regularized (ridge/\(L2\)) logistic regression to blend the predictions?
- [ ] Document the *flawed* assumptions of this ML pipeline for the historical record.

**Comment by ashutoshk314** [comment]
Hello,  
I would like to work on this issue if it is still open. Please let me know if I may proceed.

Thank you.

**Comment by eric15342335** [comment]
Hi @ashutoshk314,

Thanks for your interest in this project! To sum up my attitude toward this project, I wouldn't recommend you work on this if you are looking for typical machine learning datasets to improve your skills.

Right now, the issues are more focused on the dataset (data collection phase) itself. We only have the best (top 1) scores of each "map" (similar to 'levels' in RPGs). This discards a lot of useful information, such as skill progression, number of retries, etc.

Those to-do list points surely could improve the prediction itself (in terms of R-squared), but I believe that those are just minor nitpicks that don't address the underlying issue.

That said, I'm leaving the issue open for backup records only so that if I get more data, I can work on these improvements later.

Happy New Year, btw!

Best,
Eric

You might have heard of Kaggle already. I highly recommend you check out this newly launched competition: <https://www.kaggle.com/competitions/playground-series-s6e1>

**Comment by ashutoshk314** [comment]
Thank you for your suggestion pal!
